Adopt dynamicDowncast<> in initializer lists

annevk opened the pull request:
#### 04a8e9040886e1de0a6a5575c933d93e810e0671
<pre>
Adopt dynamicDowncast&lt;&gt; in initializer lists
<a href="https://bugs.webkit.org/show_bug.cgi?id=270091">https://bugs.webkit.org/show_bug.cgi?id=270091</a>

Reviewed by Chris Dumez.

For security &amp; performance.

Thanks to cdumez for help on the approach.

* Source/WebCore/history/CachedFrame.cpp:
(WebCore::CachedFrameBase::CachedFrameBase):
(WebCore::CachedFrameBase::initializeWithLocalFrame):
* Source/WebCore/history/CachedFrame.h:
* Source/WebCore/history/CachedPage.cpp:
(WebCore::CachedPage::CachedPage):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::m_streamDataParser):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::RenderImage):
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::TextBoxPainter&lt;TextBoxPath&gt;::TextBoxPainter):

Canonical link: <a href="https://commits.webkit.org/275326@main">https://commits.webkit.org/275326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4c0ca8c93e07ac8298c786279ea175cb8a30cd72

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41535 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20549 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43913 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44103 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37627 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43842 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23656 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17879 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34345 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42109 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17482 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35778 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15006 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/15198 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36783 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45487 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37728 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37103 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/40876 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16350 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13428 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/39284 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17969 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9311 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18025 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17613 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a8727d1f2d54fa410794f3fe105bdc95514607a5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/25105

cdumez reviewed (approved):
LGTM with fix.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4c0ca8c93e07ac8298c786279ea175cb8a30cd72. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/25105

annevk committed:
Adopt dynamicDowncast<> in initializer lists
https://bugs.webkit.org/show_bug.cgi?id=270091

Reviewed by Chris Dumez.

For security & performance.

Thanks to cdumez for help on the approach.

* Source/WebCore/history/CachedFrame.cpp:
(WebCore::CachedFrameBase::CachedFrameBase):
(WebCore::CachedFrameBase::initializeWithLocalFrame):
* Source/WebCore/history/CachedFrame.h:
* Source/WebCore/history/CachedPage.cpp:
(WebCore::CachedPage::CachedPage):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::m_streamDataParser):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::RenderImage):
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::TextBoxPainter<TextBoxPath>::TextBoxPainter):

Canonical link: https://commits.webkit.org/275326@main

webkit-commit-queue commented:
Committed 275326@main (04a8e9040886): <https://commits.webkit.org/275326@main>

Reviewed commits have been landed. Closing PR #25105 and removing active labels.
